src/storage/src/source/mysql/replication.rs

```diff
@@ -308,11 +308,9 @@ pub(crate) fn render<G: Scope<Timestamp = GtidPartition>>(
                 metrics.total.inc();
 
                 match event_data {
-                    Some(EventData::HeartbeatEvent) => {
-                        // The upstream MySQL source only emits a heartbeat when there are no other events
-                        // sent within the heartbeat interval. This means that we can safely advance the
-                        // frontier once since we know there were no more events for the last sent GTID.
-                        // See: https://dev.mysql.com/doc/refman/8.0/en/replication-administration-status.html
+                    Some(EventData::XidEvent(_)) => {
```

Comment:
Shouldn't we use the `xid` data present in this `XidEvent` to advance our capability? Or at least assert that it has the value we expect?

Reply:
The event itself just contains the `Xid` which is an identifier that has no meaning to us -- it's used to identify the transaction in the mysql 2pc protocol
https://docs.rs/mysql_common/latest/mysql_common/binlog/events/struct.XidEvent.html

Reply:
Ah ok, I thought `xid == gtid.timestamp`